### PR TITLE
Initial implementation of <MUIPlacesAutocomplete> component

### PR DESCRIPTION
**Pull request by Giners**
This commit is the initial implementation of the <MUIPlacesAutocomplete>
component. The component leverages the <Autosuggest> component from the
'react-autosuggest' package as well as the Google Places API Web
Service/Google Maps JavaScript API to provide autosuggest/completion
functionality when searching for a place. The main feature of the
<MUIPlacesAutocomplete> component is that it provides a Google Material
Design feel to the rendered elements via the Material-UI library.

When reviewing please be critical (in a constructive manner) while taking the viewpoint of someone who is going to download/consume this.

### CODE REVIEW

File: src/MUIPlacesAutocomplete.jsx [hunk 1]
```diff
@@ -1,6 +1,213 @@
...
+    // of truth for their state.
+    this.state = { value: '', suggestions: [] }
+
+    this.onSuggestionsFetchRequested = this.onSuggestionsFetchRequested.bind(this)
+    this.onSuggestionsClearRequested = this.onSuggestionsClearRequested.bind(this)
+    this.onChange = this.onChange.bind(this)
+  }
+
+  componentWillMount() {
+    this.autocompleteService = new window.google.maps.places.AutocompleteService()
```
Comment (Giners) [inline src/MUIPlacesAutocomplete.jsx:131]: Can you give guidance for how to properly consume 3rd party libraries that are expected to be loaded in the browser/window?

For example should I account for cases where the library doesn't load and if so what ought I do? Or is this fine the way it is implemented and if the library doesn't load its okay to let this component blow up and errors show up in the console?

File: test/api-key.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+// Set your Google API key so that it can be used for the tests
+export default 'Replace with your Google API key'
```
Comment (mattbailey) [inline test/api-key.js]: better done with env vars, e.g. `process.env.GOOGLE_API_KEY` so you can easily run tests in CI

**Review by mattbailey — COMMENTED** [review]
LGTM, just one comment on your api key

**Review by mattbailey — DISMISSED** [review]

File: src/MUIPlacesAutocomplete.jsx (hunk 1 above)
Comment (mattbailey) [reply src/MUIPlacesAutocomplete.jsx:131]: generally, this is hard.  You might want to see if someone has wrapped this kind of thing up before, e.g.: https://www.npmjs.com/package/react-google-maps

Comment (Giners) [reply src/MUIPlacesAutocomplete.jsx:131]: Ya. This seems like a different set of problems that I have never had to consider before.

I saw that package (and others) in the past when I was looking for something myself but thought that they weren't sufficient in terms of getting a MUI feeling applied to them. Maybe I was wrong just based on my limited knowledge.

Not looking to re-invent the wheel but it feels good to produce this myself. I just get to further learn more about the ecosystem Im working with. Plus I have open sourced something! w00t! :)

Comment (mattbailey) [reply src/MUIPlacesAutocomplete.jsx:131]: https://www.npmjs.com/package/google-maps

there ya go, someone did simply wrap it in a module with 0 changes, should check it out

Comment (mattbailey) [reply src/MUIPlacesAutocomplete.jsx:131]: otherwise, yeah, all good

**Review by mattbailey — APPROVED** [review]

File: src/MUIPlacesAutocomplete.jsx (hunk 1 above)
Comment (Giners) [reply src/MUIPlacesAutocomplete.jsx:131]: TY. Im going to make a task to look at this.

**Pull request merged**